### PR TITLE
Fix addition/git branch

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: ultralytics/yolo/utils/callbacks/mlflow.py [hunk 1]
```diff
@@ -27,9 +29,10 @@ def on_pretrain_routine_end(trainer):
         mlflow.set_tracking_uri(mlflow_location)
 
         experiment_name = '/Shared/' + trainer.args.project or '/Shared/YOLOv8'
-        experiment = mlflow.get_experiment_by_name(experiment_name)
-        if experiment is None:
-            mlflow.create_experiment(experiment_name)
```
Comment (ghost) [inline ultralytics/yolo/utils/callbacks/mlflow.py]: @plaurent the issue was here

```python
if experiment is None:
    create experiment, not get_experiment_by_name again
```

Fixed by changing it to:

```python
while (experiment := mlflow.get_experiment_by_name(experiment_name)) is None:
    experiment_name = mlflow.create_experiment(experiment_name)
```

**Comment by ghost** [comment]
You can disable the DirtyRepo check by setting the environment variable `DEVELOPMENT` to `True`

File: ultralytics/yolo/utils/callbacks/mlflow.py (hunk 1 above)
Comment (plaurent) [reply ultralytics/yolo/utils/callbacks/mlflow.py]: @YoRigid Are we at risk of an infinite loop here (if something's wrong with mlflow's connection and `.get_experiment_by_name()` always returns None?

Comment (ghost) [reply ultralytics/yolo/utils/callbacks/mlflow.py]: Good point, I didn't think about that


Let me fix it an other way

> my thought was, get experiment, if not experiment: create experiment and get the ID from create experiment then get experiment again to get the new experiment.
> but you are right, I can't guarantee creating an experiment succeeds

Comment (ghost) [reply ultralytics/yolo/utils/callbacks/mlflow.py]: @plaurent, new fix

```python
experiment = mlflow.get_experiment_by_name(experiment_name)
if experiment is None:
    mlflow.create_experiment(experiment_name)
    experiment = mlflow.get_experiment_by_name(experiment_name)
```

**Review by plaurent — APPROVED** [review]
Awesome, thank you

**Pull request merged**